connectivity/CMakeLists.txt

```diff
@@ -7,33 +7,20 @@ add_library(mbed-ble INTERFACE)
 add_library(mbed-ble-blue_nrg INTERFACE)
 add_library(mbed-ble-cordio INTERFACE)
 add_library(mbed-ble-cordio_ll INTERFACE)
-add_library(mbed-cellular INTERFACE)
```

Comment:
Does it build? I think the cellular drivers would complain with something like:
```
Cannot specify link libraries for target "mbed-cellular" which is not built by this project.
```

If not, it is perhaps because `connectivity/cellular/` is alphabetically before `connectivity/drivers/` thus the `mbed-cellular` CMake target is created by the time CMake reaches `drivers/`.
I noticed that mbed-ble in [this PR](https://github.com/ARMmbed/mbed-os/pull/14377/files) remained in this file. Is it because `connectivity/FEATURE_BLE/` comes after `connectivity/drivers/` and thus shows the error:
```
Cannot specify link libraries for target "mbed-ble" which is not built by this project.
```

Should we keep all connectivity CMake targets here until maybe we refactor the connectivity directory?

@pan- @evedon

Reply:
The thing here is that all drivers for cellular are part of mbed-cellular - how t hey were designed. there is no label ifdef even with the old tools. I'll raise multiple issues about how libraries are being built so teams can look at.

ble drivers are optional therefore I kept it as it was. Same for netsockets.

Reply:
What I meant in short is:
`mbed-ble` is defined in `connectivity/CMakeLists.txt` whilst `mbed-cellular` is defined in `connectivity/cellular/CMakeLists.txt`. Why?

Reply:
The order matters in here. I'll clean this up with comments about the order there.

Reply:
I'll rebase this once BLE gets in (it fixes this  file and the order).